### PR TITLE
(feat) gateway-nodes-enhancements

### PR DESCRIPTION
**Pull request by valcker**
- Replace GatewayNodes subscriptions with polling
- Minor Gateway Nodes page enhancements
- ESLint fixes

### CODE REVIEW

File: packages/frontend/src/model/GatewayNodes/queries/useGateways.ts [hunk 1]
```diff
@@ -16,26 +17,29 @@ type Gateway = {
...
-  }, []);
+  return data
+    .slice()
+    .sort((a, b) => b.lastPingAt - a.lastPingAt)
+    .reduce((acc: any[], gateway: Gateway, idx: number) => {
+      const lastPingFormatted = formatDistanceToNow(gateway.lastPingAt, { addSuffix: true });
+      // We can mutate acc because we have new [] each time we call this function
+      acc[idx] = [gateway.nodeId, gateway.hostname, gateway.status, lastPingFormatted, gateway.configTimestamp];
+      return acc;
+    }, []);
```
Comment (Feshchenko) [inline packages/frontend/src/model/GatewayNodes/queries/useGateways.ts:35]: Можно использовать деструктуризацию и убрать лишнюю переменную
```
.reduce((acc: any[], { nodeId, hostname, status, lastPingAt, configTimestamp }: Gateway, idx: number) => {
  acc[idx] = [nodeId, hostname, status, formatDistanceToNow(lastPingAt, { addSuffix: true }), configTimestamp];
  return acc;
}, [])
```

File: packages/frontend/src/model/GatewayNodes/queries/useGateways.ts [hunk 2]
```diff
@@ -16,26 +17,29 @@ type Gateway = {
...
 const createNodeList = (data: Gateway[]) => {
   if (!data) return [];
 
-  return data.reduce((acc: any[], gateway: Gateway, idx: number) => {
-    // We can mutate acc because we have new [] each time we call this function
-    acc[idx] = [gateway.nodeId, 'active', gateway.lastPingAt, gateway.configTimestamp];
-    return acc;
-  }, []);
+  return data
+    .slice()
```
[the remark this answers is not in the document]
Comment (valcker) [reply packages/frontend/src/model/GatewayNodes/queries/useGateways.ts:28]: `data` is an immutable array, `slice()` is needed to create a copy of it.

File: packages/backend/src/common/enum/gateway-status-timers.enum.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+enum GatewayStatusTimers {
+  REMOVE_TIMEOUT = 300_00,
```
Comment (germankrause) [inline packages/backend/src/common/enum/gateway-status-timers.enum.ts:2]: 30_000?

File: packages/backend/src/modules/gateway/gateway.service.ts [hunk 4]
```diff
@@ -23,8 +24,8 @@ export default class GatewayService {
   }
 
   private onPing(data: string): void {
-    const { nodeId, configTimestamp } = JSON.parse(data);
-    this.nodes[nodeId] = { nodeId, lastPingAt: Date.now(), configTimestamp };
+    const { nodeId, configTimestamp, hostname } = JSON.parse(data);
+    this.nodes[nodeId] = { nodeId, lastPingAt: Date.now(), configTimestamp, hostname, status: 'active' };
```
Comment (germankrause) [inline packages/backend/src/modules/gateway/gateway.service.ts:28]: what about `GatewayStatusEnum = { ACTIVE, IDLE }`?

File: packages/frontend/src/model/GatewayNodes/queries/useGateways.ts [hunk 5]
```diff
@@ -16,26 +17,34 @@ type Gateway = {
...
 export const useGateways = () => {
-  const { data, loading, error, subscribeToMore } = useQuery(QUERY_GATEWAYS);
-
-  subscribeToMore({
-    document: GATEWAYS_UPDATE,
-    updateQuery: (_, { subscriptionData }) => {
-      return { getGateways: createNodeList(subscriptionData.data.gatewaysUpdated) };
-    },
+  // random interval between 5 and 20 secs
+  const randomPollInterval = (Math.floor(Math.random() * 16) + 5) * 1000;
```
Comment (germankrause) [inline packages/frontend/src/model/GatewayNodes/queries/useGateways.ts:45]: what was the problem with the subscription? should we remove it from the backend?

File: packages/backend/src/common/enum/gateway-status-timers.enum.ts (hunk 3 above)
Comment (valcker) [reply packages/backend/src/common/enum/gateway-status-timers.enum.ts:2]: A typo.

File: packages/frontend/src/model/GatewayNodes/queries/useGateways.ts (hunk 5 above)
Comment (valcker) [reply packages/frontend/src/model/GatewayNodes/queries/useGateways.ts:45]: There was a problem with authorization and passing an accessToken in the subscription itself. At the moment, the backend doesn't work at all if we are trying to do that. Besides, Apollo themselves do not recommend to use the subscriptions _unless_ the amount of data is important (which is not in our case).

**Pull request merged**